src/cursor/cur_index.c

```diff
@@ -210,11 +240,16 @@ __curindex_search(WT_CURSOR *cursor)
 	 * otherwise the primary key columns will be appended to the index key,
 	 * but we don't disallow that (odd) case.
 	 */
-	if (exact < 0)
+	if (cmp < 0)
 		WT_ERR(child->next(child));
 
-	if (child->key.size < cursor->key.size ||
-	    memcmp(child->key.data, cursor->key.data, cursor->key.size) != 0) {
+	if (child->key.size < cursor->key.size)
+		WT_ERR(WT_NOTFOUND);
```

Comment:
@michaelcahill, the `size` test looks correct to me, but I had to stare at it a little. Is it worth special-casing? Why not always call `__wt_compare` and let it fail?